test/user_test.dart

```diff
@@ -116,4 +116,299 @@ Future<void> main([List<String>? args]) async {
+
+    await user.apiKeys.disable(first.id);
+
+    final fetched = await user.apiKeys.fetchAll();
+    expect(fetched[0].id, first.id);
+    expect(fetched[0].isEnabled, false);
+
+    expect(fetched[1].id, second.id);
+    expect(fetched[1].isEnabled, true);
+
+    await user.apiKeys.enable(first.id);
+
+    final refetched = await user.apiKeys.fetchAll();
+    expect(refetched[0].id, first.id);
+    expect(refetched[0].isEnabled, true);
+
+    expect(refetched[1].id, second.id);
+    expect(refetched[1].isEnabled, true);
+  });
+
+  baasTest('User.apiKeys can login with generated key', (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+
+    final key = await user.apiKeys.create('my-key');
+    final credentials = Credentials.apiKey(key.value!);
+
+    final apiKeyUser = await app.logIn(credentials);
+    expect(apiKeyUser.provider, AuthProviderType.apiKey);
+    expect(apiKeyUser.id, user.id);
+    expect(apiKeyUser.refreshToken, isNot(user.refreshToken));
+  });
+
+  baasTest('User.apiKeys can login with reenabled key', (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+
+    final key = await user.apiKeys.create('my-key');
+
+    await user.apiKeys.disable(key.id);
+
+    final credentials = Credentials.apiKey(key.value!);
+
+    await expectLater(
+        () => app.logIn(credentials),
+        throwsA(isA<AppException>()
+            .having((e) => e.message, 'message', contains('invalid API key'))
+            .having((e) => e.statusCode, 'statusCode', 401)
+            .having((e) => e.linkToServerLogs, 'linkToServerLogs', contains('logs?co_id='))));
+
+    await user.apiKeys.enable(key.id);
+
+    final apiKeyUser = await app.logIn(credentials);
+    expect(apiKeyUser.provider, AuthProviderType.apiKey);
+    expect(apiKeyUser.id, user.id);
+    expect(apiKeyUser.refreshToken, isNot(user.refreshToken));
+  });
+
+  baasTest("User.apiKeys can't login with disabled key", (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+
+    final key = await user.apiKeys.create('my-key');
+
+    await user.apiKeys.disable(key.id);
+
+    final credentials = Credentials.apiKey(key.value!);
+
+    await expectLater(
+        () => app.logIn(credentials),
+        throwsA(isA<AppException>()
+            .having((e) => e.message, 'message', contains('invalid API key'))
+            .having((e) => e.statusCode, 'statusCode', 401)
+            .having((e) => e.linkToServerLogs, 'linkToServerLogs', contains('logs?co_id='))));
+  });
+
+  baasTest("User.apiKeys can't login with deleted key", (configuration) async {
+    final app = App(configuration);
+    final user = await getIntegrationUser(app);
+
+    final key = await user.apiKeys.create('my-key');
+
+    await user.apiKeys.delete(key.id);
+
+    final credentials = Credentials.apiKey(key.value!);
```

Comment:
Isn't it strange for the API that `Credentials.apiKey(key.value!)` requires apiKey but we have to pass its value which is possible to be null?

Reply:
We don't have a test for login with an `apiKey` after it has been received with `fetrch`. Will we have the `value` populated then? I can check it, it is just a reminder.

Reply:
No, you can't fetch an API key value after it was created. The idea is that you create an API key and you store its value somewhere, then you use it to login. It's not expected that you have the full API key at this point, only the value.

Reply:
Of course, this is for security reasons. I didn't think about that.